appinventor/components/src/com/google/appinventor/components/runtime/SpeechRecognizer.java

```diff
@@ -71,6 +74,29 @@ public SpeechRecognizer(ComponentContainer container) {
     UseLegacy(useLegacy);
   }
 
+  /**
+   * Suggests the language to use for recognizing speech. An empty string (the default) will
+   * use the system's default language.
+   *
+   *     Language is specified using an IETF language tag, such as en-US or es-MX.
+   *
+   * @return the target language for recognition
+   */
+  @SimpleProperty(category = PropertyCategory.BEHAVIOR)
```

Comment:
Should this be a designer property?

Reply:
I punted on this since we'd either want to have a dropdown (there are 183 active language codes without any further specialization) or at least a regex match to ensure the value "looks" right. We could also consider in the future adding a block to ask the speech recognizer what languages it supports. For now, I expect most people will be satisfied using the device default and will only ever need to apply this as a workaround for language specific issues.